### DISCUSSION TITLE
Support PEP 692 completions

### DISCUSSION THREAD

**Discussion by multimeric** [microsoft/pylance-release#3046] [Enhancement]
When defining a function whose `**kwargs` have statically typed elements, as defined in PEP 692, ideally these statically typed arguments would be suggested as arguments when calling that function.

For example, if I define `foo` as follows:
```python
from typing import TypedDict
from typing_extensions import Unpack

class Kwargs(TypedDict):
    a: int
    b: str
    
def foo(arg: str, **kwargs: Unpack[Kwargs]):
    pass
```

I would hope for pylance to suggest `a` and `b` as optional kwargs, but it does not:

![image](https://user-images.githubusercontent.com/5019367/178912029-7c643dac-d94c-49a0-b95e-8a272e78dfe5.png)

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by ianhco** [comment]
I believe this is a valid enhancement as currently many functions and classes are using `**kwargs`, and users have to scrape the documentation multiple times just to get the needed keywords.

**Comment by rchiodo** [comment]
This should already be working. You just need to have arguments that have more than a single character (or hit CTRL+SPACE):

![image](https://github.com/microsoft/pylance-release/assets/19672699/c481d2c4-017e-49e9-8d80-83a0a025a9d8)

**Reply by ianhco** [reply]
Thanks for the reply, however I wish to use it on classes with generics, this is what is currently being shown
![image](https://github.com/microsoft/pylance-release/assets/58193352/9dfa0489-c6af-47c7-b92b-093571a2b27f)

**Reply by erictraut** [reply]
PEP 692 does not work with generics. The `Unpack` special form can be used only with a TypedDict, not a type variable.
